src/components/card/search-hit-resource-card.tsx

```diff
@@ -33,11 +35,32 @@ const SearchHitResourceCard: React.FC<{
   ...props
 }) => {
   const {fontSize, ref} = useFitText()
+  const progressQuery = useQuery(
+    `progress:${resource?.path}`,
+    async (path) => {
+      const result = resource
+        ? await axios.get(
+            `${process.env.NEXT_PUBLIC_AUTH_DOMAIN}/api/v1${resource.path}/progress`.replace(
+              'courses',
+              'playlists',
+            ),
+          )
+        : {data: false}
+      return result.data?.progress
+    },
+    {
+      retry: false,
+    },
+  )
+
```

Comment:
clunky, but works

it's particularly interesting that if you use the same key it caches very nicely so this would probably fit best in a `useResourceProgress` hook that could manage all the edge cases and such